File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        Map<Character, Integer> m = new HashMap<>();
        int[] answer = new int[s.length()];
        for(int i = 0; i < s.length(); i++){
            if (m.containsKey(s.charAt(i))) answer[i] = (i - m.get(s.charAt(i)));
            else answer[i] = -1;
            m.put(s.charAt(i), i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let m = new Map();
    let answer = [];
    for(let i = 0; i < s.length; i++){
        if (m.has(s[i])) answer.push(i - m.get(s[i]));
        else answer.push(-1);
        m.set(s[i], i);
    }
    return answer;
}